### DISCUSSION TITLE
Refetching When Global State Changes

### DISCUSSION THREAD

**Discussion by davidulman** [reduxjs/redux-toolkit#3561] [Q&A]
Hey,

I've seen the releted question but I was wondring if there is easier way to refetch data when the global state changes. 
Basicly I want that getNebenkosten will refetch when currentObject state is changing is there a way to achive it without using queryFn in getNebenkosten? 

Here is my API slice :
`
              
    export const nebenkostenApi = createApi({
      reducerPath: 'nebenkostenApi',
      baseQuery: fetchBaseQuery({
        baseUrl: 'myapi/',
        prepareHeaders(headers, { getState }) {
          const token = getState().auth.userInformation.token;
          const profile = getState().ui.uiObject.currentObject;
          let currentPorofile = profile;
          if (token) {
            headers.set('authorization', `Bearer ${token}`);
            headers.set('profile', profile);
            headers.set('Accept', 'application/json');
          }
    
          return headers;
        },
      }),
    
      endpoints: (builder) => ({
        getNebenkosten: builder.query({
          query: () => 'nebenkosten/all',
          providesTags: ['Nebenkosten'],
        }),
    
        getNebenkostenById: builder.query({
          query: (id) => `nebenkosten/${id}`,
        }),
    
        createNebenkosten: builder.mutation({
          query: (body) => ({
            url: 'nebenkosten/',
            method: 'POST',
            body,
          }),
          invalidatesTags: ['Nebenkosten'],
        }),
    
        updateNebenkosten: builder.mutation({
          query: (body) => ({
            url: 'nebenkosten/',
            method: 'PATCH',
            body,
          }),
          invalidatesTags: ['Nebenkosten'],
        }),
    
        deleteNebenkosten: builder.mutation({
          query: (id) => ({
            url: `nebenkosten/delete/${id}`,
            method: 'DELETE',
          }),
          invalidatesTags: ['Nebenkosten'],
        }),
      }),
    });
    
    export const {
      useGetNebenkostenQuery,
      useGetNebenkostenByIdQuery,
      useCreateNebenkostenMutation,
      useUpdateNebenkostenMutation,
      useDeleteNebenkostenMutation,
    } = nebenkostenApi;
`

Thanks In Advance ! 😄 😃 
David

**Comment by davidulman** [comment]
Found while reading the api the use of the serializeQueryArgs, and its works like I wanted.
Is there better way ?

`

      endpoints: (builder) => ({
          getNebenkosten: builder.query({
            query: () => 'nebenkosten/all',
            serializeQueryArgs: (profile) => {
              return {
                profile,
              };
            },
            providesTags: ['Nebenkosten'],
          }),

`

**Reply by phryneas** [reply]
This doesn't have anything to do with refetching on global state change, so I'm very confused

**Comment by phryneas** [comment]
You could add a tag and invalidate that from a listenerMiddleware.
